lib/msal-core/samples/react-sample-app/src/App.js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import "./App.css";
+
+import AuthProvider from "./AuthProvider";
+
+class App extends React.Component {
+    render() {
+        return (
+            <div>
+                <section>
+                    <p>
+                        Welcome to the Microsoft Authentication Library For
+                        Javascript - React Quickstart
+                    </p>
+                    {!this.props.account ? (
+                        <button onClick={this.props.signIn}>Sign In</button>
+                    ) : (
+                        <>
+                            <button onClick={this.props.signOut}>
+                                Sign Out
+                            </button>
+                            <button onClick={this.props.readMail}>
+                                Read Email
+                            </button>
+                        </>
+                    )}
+                </section>
+                <section>
+                    {this.props.account && (
+                        <pre>{JSON.stringify(this.props.account, null, 4)}</pre>
+                    )}
+                    {this.props.graphData && (
+                        <pre>
+                            {JSON.stringify(this.props.graphData, null, 4)}
+                        </pre>
+                    )}
+                </section>
+            </div>
+        );
+    }
+}
+
+export default AuthProvider(App);
```

Comment:
I wonder if Hocs are new enough to most that something like this might be more descriptive of what `AuthProvider` is 

```js
export default () => (
    <AuthProivder>
      <App /> 
     </AuthProvider>
);
```

Reply:
Since we're passing down props from the provider, I don't think this will work. I would have to use context, or do render props, neither of which I'm a fan of for this sample.